tests/openapi/test_turbo4_storage.py

```
@@ -236,3 +238,872 @@ def test_quantized_turbo4_search(quant_collection_name, vector_name, rescore):
+# Vectors come back dequantized (and normalized for Cosine), so checks
+# against the written input are cosine-similarity based, never exact.
+# Re-reads of the same stored bytes, however, must match exactly — turbo4
+# quantizes once at ingest, so any drift between reads means an extra
+# dequantize→requantize pass slipped in. These tests run after the search
+# tests above (pytest definition order) and mutate disjoint point ids, so the
+# shared collection stays valid for everything else.
+###############################################################################
+
+
+def _cosine_sim(a, b):
+    dot = sum(x * y for x, y in zip(a, b))
```

Ruff (0.15.21)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)